### PR TITLE
Add the functionality of "date1 to date2 by period"

### PR DESCRIPTION
**Pull request by pishen**
Just added a function that we use a lot, the usage is as below:
```
scala> val start = LocalDate.parse("2016-01-01")
start: java.time.LocalDate = 2016-01-01

scala> val end = LocalDate.parse("2016-01-05")
end: java.time.LocalDate = 2016-01-05

scala> start to end
res1: jp.ne.opt.chronoscala.DateInterval = DateInterval(2016-01-01, 2016-01-02, 2016-01-03, 2016-01-04, 2016-01-05)

scala> start to end by 2.days
res2: jp.ne.opt.chronoscala.DateInterval = DateInterval(2016-01-01, 2016-01-03, 2016-01-05)

scala> (start to end by 2.days).map(_ + 1.day)
res3: Seq[java.time.LocalDate] = List(2016-01-02, 2016-01-04, 2016-01-06)
```

If it's possible, please give me suggestions on how to make this merged. Thanks!

### CODE REVIEW

File: src/main/scala/jp/ne/opt/chronoscala/DateInterval.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package jp.ne.opt.chronoscala
+
+import java.time.{LocalDate, Period}
+import jp.ne.opt.chronoscala.Imports.richLocalDate
```
Comment (ocadaruma) [inline src/main/scala/jp/ne/opt/chronoscala/DateInterval.scala]: [IMO] I prefer to use wildcard import (`Imports._`) to enable implicits.

File: src/main/scala/jp/ne/opt/chronoscala/DateInterval.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Represents an immutable date interval
+ */
+case class DateInterval(startDate: LocalDate, endDate: LocalDate, step: Period) extends Seq[LocalDate] {
+  def apply(idx: Int) = {
+    if (0 <= idx && idx < length) {
+      iterator.drop(idx).next
+    } else {
+      throw new IndexOutOfBoundsException()
```
Comment (ocadaruma) [inline src/main/scala/jp/ne/opt/chronoscala/DateInterval.scala]: [IMO] How about include `idx` in exception message ? (like `Seq(1, 2, 3)(4)`)

**Comment by ocadaruma** [comment]
Thank you for PR !
This feature looks very useful.
LGTM

**Comment by pishen** [comment]
Cool, thanks! Looking forward to the next release.

**Pull request merged**